pkg/cmd/target/completion.go

```diff
@@ -0,0 +1,130 @@
+/*
+SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package target
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/gardener/gardenctl-v2/internal/util"
+	"github.com/gardener/gardenctl-v2/pkg/target"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+func validArgsFunction(f util.Factory, o *Options, args []string, toComplete string) ([]string, error) {
+	if len(args) == 0 {
+		return []string{
+			string(TargetKindGarden),
+			string(TargetKindProject),
+			string(TargetKindSeed),
+			string(TargetKindShoot),
+		}, nil
+	}
+
+	kind := TargetKind(strings.TrimSpace(args[0]))
+	if err := validateKind(kind); err != nil {
+		return nil, err
+	}
+
+	manager, err := f.Manager()
+	if err != nil {
+		return nil, err
+	}
+
+	// NB: this uses the DynamicTargetProvider from the root cmd and
+	// is therefore aware of flags like --garden; the goal here is to
+	// allow the user to type "gardenctl target --garden [tab][select] --project [tab][select] shoot [tab][select]"
```

Comment:
allow to target also only with flags
`gardenctl target --garden [tab][select] --project [tab][select] --shoot [tab][select]`
`Error: failed to complete command options: expected exactly 2 arguments`